src/endpoints/crosschain/ethereum/resolve.rs

```diff
@@ -134,84 +152,200 @@ pub async fn handler(State(state): State<Arc<AppState>>, query: Query) -> impl I
-                                            return get_error(format!(
-                                                "Error signing message : {}",
-                                                e
-                                            ));
+                                        None => {
+                                            // we will just query evm-address field
+                                            match get_user_data(
+                                                &provider,
+                                                state.conf.contracts.starknetid,
+                                                id,
+                                                *EVM_ADDRESS,
+                                            )
+                                            .await
+                                            {
+                                                Some(addr) => {
+                                                    let hex_addr = to_eth_hex(&addr);
+                                                    let trimmed_hex_addr =
+                                                        hex_addr.trim_start_matches("0x");
+                                                    let bytes = ethers::utils::hex::decode(
+                                                        trimmed_hex_addr,
+                                                    )
+                                                    .map_err(|err| {
+                                                        get_error(format!(
+                                                            "Invalid Structure: {}",
+                                                            err
+                                                        ))
+                                                    })
+                                                    .unwrap();
+                                                    vec![Token::Bytes(bytes)]
+                                                }
+                                                None => {
+                                                    return get_error(
+                                                        "No evm address specified for this domain"
+                                                            .to_string(),
+                                                    );
+                                                }
+                                            }
                                         }
                                     }
                                 }
+                            } else {
+                                return get_error(format!("Chain not implemented: {}", chain));
                             }
-                            Err(e) => {
-                                println!("Error fetching user data for field {} : {}", field, e)
+                        }
+                        ResolverFunctionCall::Addr(_bf) => {
```

Comment:
Just to make sure I understand: this one is called for native resolving on Ethereum, while `ResolverFunctionCall::AddrMultichain` is called for all other chains, including evm ones?

Reply:
Yes from my understanding and testing, that's how it works